sdk/cosmos/azure-cosmos/src/test/java/com/azure/cosmos/CosmosMultiHashTest.java

```diff
@@ -170,42 +195,120 @@ private void validateDocCRUDandQuery() throws Exception {
+        wrongDoc.set("id", new TextNode(UUID.randomUUID().toString()));
+        wrongDoc.set("city", new TextNode("Redmond"));
+        try {
+            createdMultiHashContainer.createItem(wrongDoc);
+        } catch (CosmosException e) {
+            assertThat(e.getStatusCode()).isEqualTo(400);
+            assertThat(e.getMessage().contains("PartitionKey extracted from document doesn't match the one specified in the header.")).isTrue();
         }
 
         //Document Read
-        {
-            for (int i = 0; i < docs.size(); i++) {
-                ObjectNode doc_current = docs.get(i);
-                PartitionKey partitionKey = new PartitionKeyBuilder()
-                    .add(doc_current.get("city").asText())
-                    .add(doc_current.get("zipcode").asText())
-                    .build();
-                CosmosItemResponse<ObjectNode> response = createdMultiHashContainer.readItem(doc_current.get("id").asText(), partitionKey, ObjectNode.class);
-            }
+        for (int i = 0; i < docs.size(); i++) {
+            ObjectNode doc_current = docs.get(i);
+            partitionKey = new PartitionKeyBuilder()
+                .add(doc_current.get("city").asText())
+                .add(doc_current.get("zipcode").asText())
+                .build();
+            CosmosItemResponse<ObjectNode> response = createdMultiHashContainer.readItem(doc_current.get("id").asText(), partitionKey, ObjectNode.class);
+            validateItemResponse(doc_current, response);
+        }
+
+        //Document Read - Negative tests
+        //Using incomplete partition key
+        PartitionKey partialPK = new PartitionKeyBuilder().add(doc.get("city").asText()).build();
+        try {
+            createdMultiHashContainer.readItem(doc.get("id").asText(), partialPK, ObjectNode.class);
+        } catch (CosmosException e) {
+            assertThat(e.getStatusCode()).isEqualTo(400);
+            assertThat(e.getMessage().contains("Partition key provided either doesn't correspond to definition in the collection or doesn't match partition key field values specified in the document.")).isTrue();
         }
 
-        // Query Tests.
+        //Read Many
+        List<Pair<String, PartitionKey>> pairList = new ArrayList<>();
+        List<CosmosItemIdentity> itemList = new ArrayList<>();
+        for (ObjectNode jsonNodes : docs) {
+            pairList.add(Pair.of(jsonNodes.get("id").asText(),
+                new PartitionKeyBuilder().add(jsonNodes.get("city").asText()).add(jsonNodes.get("zipcode").asText()).build()));
+            PartitionKey pkToUse = new PartitionKeyBuilder().add(jsonNodes.get("city").asText()).add(jsonNodes.get("zipcode").asText()).build();
+            itemList.add(new CosmosItemIdentity(pkToUse, jsonNodes.get("id").asText()));
+        }
+        FeedResponse<ObjectNode> documentFeedResponse = createdMultiHashContainer.readMany(itemList, ObjectNode.class);
```

Comment:
actually this brings a good discussion point: 

For readMany, SDK internally will try to group by physical partitions. if there is only one item in the partition bucket, then we do readItem instead of query. however, for read item, we will need the full partition key. So it will fail. 

I wonder whether we should add the following logic:
if the partition key is partial partition key, even there is only one item in the partition bucket, we should still issue a query instead read.

Reply:
Discussed offline, readMany should allow full Pk, throw if any partition PK being defined